extensions/pkg/controller/healthcheck/reconciler.go

```diff
@@ -123,6 +123,11 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 				return r.resultWithRequeue(), err
```

Comment:
Why not 
```suggestion
				return reconcile.Result{}, err
```
here?

Reply:
Because it failed to update the extension conditions to "hibernated" - it should do that before stopping the health checks. This is to show that it stopped the health-checks deliberately due to hibernation.

Reply:
As the error is returned then it should be requeued automatically anyways (with exponential backoff instead of fixed delay), or? Am I missing something?

Reply:
missed that we are returning the error too - will adapt.